app/markets/ActivityTable.tsx

```diff
@@ -48,67 +49,89 @@ export const ActivityTable = ({ id }: { id: string }) => {
-        id: id.toLowerCase(),
+        skip: (page - 1) * ITEMS_PER_PAGE,
+        fpmm: id,
+        orderBy: FpmmTrade_OrderBy.CreationTimestamp,
+        orderDirection: OrderDirection.Desc,
       }),
   });
 
-  const hasMoreMarkets =
-    transactionsNextPage && transactionsNextPage.fpmmTransactions.length !== 0;
+  const { data: tradesNextPage } = useQuery({
+    queryKey: ['getMarketTrades', id, page],
+    queryFn: async () =>
+      getMarketTrades({
+        first: ITEMS_PER_PAGE,
+        skip: page * ITEMS_PER_PAGE,
+        fpmm: id,
+        orderBy: FpmmTrade_OrderBy.CreationTimestamp,
+        orderDirection: OrderDirection.Desc,
+      }),
+  });
 
+  const hasMoreMarkets = tradesNextPage && tradesNextPage.fpmmTrades.length !== 0;
   const showPaginationButtons = hasMoreMarkets || page !== 1;
-
-  const activities = data?.fpmmTransactions ?? [];
+  const tradeActivities = trades?.fpmmTrades;
+  const txsActivities = txs?.fpmmTransactions;
 
   return (
     <div>
       <Table>
         <TableHeader>
           <TableRow>
-            <TableHead className="pl-4 text-text-low-em">Users</TableHead>
+            <TableHead className="pl-4 text-text-low-em">User</TableHead>
             <TableHead className="text-text-low-em">Action</TableHead>
             <TableHead className="text-text-low-em">Shares</TableHead>
             <TableHead className="pr-4 text-right text-text-low-em">Date</TableHead>
           </TableRow>
         </TableHeader>
-        {isLoading ? (
+        {isTxsLoading || isTradesLoading ? (
           <LoadingSkeleton />
         ) : (
           <TableBody className="text-base font-semibold">
-            {activities.map(activity => (
-              <TableRow key={activity.transactionHash}>
-                <TableCell className="pl-4 text-text-high-em">
-                  <a
-                    href={getGnosisAddressExplorerLink(activity.user.id)}
-                    className="hover:underline"
-                    target="_blank"
-                    rel="noopener noreferrer"
-                  >
-                    {shortenAddress(activity.user.id)}
-                  </a>
-                </TableCell>
-                <TableCell>
-                  <Tag
-                    size="xs"
-                    colorScheme={getTagColorScheme(activity.transactionType)}
-                    className="w-fit uppercase"
-                  >
-                    {activity.transactionType}
-                  </Tag>
-                </TableCell>
-                <TableCell className="truncate text-text-high-em">
-                  {formatEtherWithFixedDecimals(activity.sharesOrPoolTokenAmount)}
-                </TableCell>
-                <TableCell className="text-nowrap pr-4 text-right text-text-low-em">
-                  {formatDateTime(activity.creationTimestamp)}
-                </TableCell>
-              </TableRow>
-            ))}
+            {txsActivities?.map(activity => {
```

Comment:
Here we should only iterate through trades activities

Reply:
Done!